File: DKImagePickerControllerDemo_OC/DKImagePickerControllerDemo_OC/Source/DKImagePickerController/View/Cell/DKAssetGroupDetailCameraCell.m
//
//  DKAssetGroupDetailCameraCell.m
//  DKImagePickerControllerDemo_OC
//
//  Created by 赵铭 on 2017/6/26.
//  Copyright © 2017年 zm. All rights reserved.
//

#import "DKAssetGroupDetailCameraCell.h"
#import "DKImageResource.h"
@implementation DKAssetGroupDetailCameraCell

+ (NSString *)cellReuseIdentifier{
    return @"DKImageCameraIdentifier";
}

- (instancetype)initWithFrame:(CGRect)frame{
    if ( self = [super initWithFrame:frame]) {
        UIImageView * cameraImageView = [[UIImageView alloc] initWithFrame:self.bounds];
        cameraImageView.contentMode = UIViewContentModeCenter;
        cameraImageView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
        cameraImageView.image = [DKImageResource cameraImage];
        
        [self.contentView addSubview:cameraImageView];
        self.contentView.backgroundColor = [UIColor colorWithWhite:0.9 alpha:1.0];
    }
    return self;
}
/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: DKImagePickerController/View/Cell/DKAssetGroupDetailCameraCell.swift
//
//  DKAssetGroupDetailCameraCell.swift
//  DKImagePickerController
//
//  Created by ZhangAo on 07/12/2016.
//  Copyright © 2016 ZhangAo. All rights reserved.
//

import UIKit

class DKAssetGroupDetailCameraCell: DKAssetGroupDetailBaseCell {
    
    class override func cellReuseIdentifier() -> String {
        return "DKImageCameraIdentifier"
    }
    
    private var cameraImageView: UIImageView?
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        let cameraImageView = UIImageView(frame: self.bounds)
        cameraImageView.contentMode = .center
        cameraImageView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        self.contentView.addSubview(cameraImageView)
        self.cameraImageView = cameraImageView
        
        self.contentView.backgroundColor = UIColor(white: 0.9, alpha: 1.0)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override var imagePickerController: DKImagePickerController? {
        willSet {
            if let newValue = newValue, let cameraImageView = self.cameraImageView, cameraImageView.image == nil {
                cameraImageView.image = newValue.imageResource.cameraImage()
            }
        }
    }
    
} /* DKAssetGroupDetailCameraCell */
